discord_quote/discord_quote/discord_quote.py

```diff
@@ -634,6 +651,72 @@ def pred(m):
                          'invalid_situation_request',
                           ctx.message.channel.name]))
 
+@bot.command()
+async def test(ctx):
+    # Function for debugging the current status of all the quote commands.
+
+    # --- Helper functions ---
+    async def get_last_real_message():
+        # Get the last message in the channel from a real person.
+        # We need to skip the initial !test request.
+        counter = 0
+        async for elem in ctx.channel.history():
+            if not elem.author.bot:
+                counter += 1
+                if counter > 1:
+                    return(elem)
+
+    async def get_last_message():
+        # Helper function to test quoting quotes
+        # Get the last message in the channel (not including the request)
+        messages = await ctx.channel.history(limit=2).flatten()
+        return(messages[1])
+
+    # --- Tests ---
+    # Get the last real message
+    msg_ = await get_last_real_message()
+
+    # Grab the command
+    quote_cmd = ctx.bot.get_command('quote')
+
+    await ctx.channel.send('|---TESTING QUOTE FUNCTIONS---|')
+    # TEST 1: Quote without a reply
+    await ctx.invoke(quote_cmd, request=f'{msg_.id}')
+
+    # TEST 2: Quote with a reply
+    await ctx.invoke(quote_cmd
+            , request=f'{msg_.id} Testing quote + reply functionality.')
+
+    # TEST 3: Quote a quote without a reply
+    msg_ = await get_last_message()
+    await ctx.invoke(quote_cmd, request=f'{msg_.id}')
+
+    # TEST 4: Quote a quote with a reply
```

Comment:
These seem like they could go slightly wrong if the bot gets interrupted by another message, but it's probably too marginal to worry about; I assume this whole test sequence runs very fast.